File: happy-number/happy-number.py
#leetcode - happy number - easy

class Solution:
    def sumOfSquares(self, num):
        result = 0
        while num > 0:
            result += (num % 10)**2
            num = num // 10
        return result

    def isHappy(self, n: int) -> bool:
        seen = set()
        sum = self.sumOfSquares(n)
        seen.add(n)
        while sum != 1:
            if sum in seen:
                return False
            seen.add(sum)
            sum = self.sumOfSquares(sum)
        return True

File: happy-number/happy_number.js
//leetcode - happy number - easy

var sumOfSquares = function(num) {
    let result = 0;
    while (num > 0) {
        result += (num % 10)**2;
        num = Math.floor(num / 10)
    }
    return result
};

var isHappy = function(n) {
    const seen = new Set();
    seen.add(n);
    let sum = sumOfSquares(n);
    while (sum !== 1) {
        if (seen.has(sum)) {
            return false
        } else {
            seen.add(sum)
        }
        sum = sumOfSquares(sum)
    };
    return true
};